Admin tag pages

Allcharles opened the pull request:
# Admin Tag Pages

Created admin tag pages (list, new, edit, delete).

## Changes

- Created `TagsService`
- Created `Tag` model
- Created `FormlyCheckboxInput`
- Created `FormlyHorizontalWrapper`
- Created tag list/forms

## Issues

- Update forms to provide `typeOfTags` dropdown
  - Create custom resolver and service function
- Missing unit tests

## Visual Changes

Tag list page
![image](https://user-images.githubusercontent.com/3955116/77750249-2e60d480-706f-11ea-82a4-8fd622464a81.png)

New Tag Form
![image](https://user-images.githubusercontent.com/3955116/77812311-36ab2500-70ec-11ea-9670-94c95250b12b.png)


Edit Tag Form
![image](https://user-images.githubusercontent.com/3955116/77812347-62c6a600-70ec-11ea-9571-1039df60d706.png)

Delete Tag Form
![image](https://user-images.githubusercontent.com/3955116/77812351-69551d80-70ec-11ea-87a3-c4ea08ca67a0.png)

Allcharles committed:
Created Tag/Tag Service

Allcharles committed:
Created tag pages

Allcharles committed:
Fix tag list component

Allcharles committed:
Created checkbox component

Allcharles committed:
Created tag forms

Allcharles committed:
Created formly checkbox input

- Wrapped checkbox in bootstrap custom checkbox
- Created formly horizontal wrapper to allow inlining of form inputs

Allcharles committed:
Added showMock function

atruskie commented on src/app/models/Tag.ts:
```diff
@@ -0,0 +1,77 @@
+      typeOfTag: this.typeOfTag,
+      retired: this.retired,
+      notes: this.notes,
+      creatorId: this.creatorId,
+      updaterId: this.updaterId,
+      createdAt: this.createdAt?.toISO(),
+      updatedAt: this.updatedAt?.toISO()
```
These will only ever be updated by the server. Remove from toJSON

Allcharles replied:
Extracted to new issue: https://github.com/QutEcoacoustics/workbench-client/issues/173

atruskie commented on src/app/models/Tag.ts:
```diff
@@ -0,0 +1,77 @@
+    };
+  }
+
+  redirectPath(): string {
```
not central to this issue, but the naming of `redirectPath` is a little weird. It implies I can only use that function will I am doing a redirect... I can't just use it navigate to the view?

Allcharles replied:
Extracted to new issue: https://github.com/QutEcoacoustics/workbench-client/issues/173

atruskie commented on src/app/services/baw-api/tags.service.ts:
```diff
@@ -0,0 +1,64 @@
+    isTaxanomic: modelId % 5 === 0,
+    typeOfTag: "general",
+    retired: modelId % 5 === 0,
+    notes: "PLACEHOLDER NOTES",
```
anywhere you see `notes` on one of our models, it should always returns an object/hash - it'll be a json blob in the database that gets inserted into the model's notes field

Allcharles replied:
Extracted to new issue: https://github.com/QutEcoacoustics/workbench-client/issues/173

Allcharles committed:
Admin tag service (#169)

- Extract Resolvers to generic class
- Created TypeOfTagsResolver
- Implement tag service resolvers
- Extract TagType to class
- Fix tag service endpoint
  - Endpoint was missing /new and /filter extensions to endpoint

Allcharles committed:
Merge branch 'master' into admin-tag-pages

Allcharles committed:
Remove redundant showMock

Allcharles committed:
Added unit tests
